.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: ci
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  automated-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php_version: ["8.3"]
+        core_version: ["10.4.0"]
+    env:
+      PHP_VERSION: ${{ matrix.php_version }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Modify web container version
+        run: |
+          # Set the php version, and replace -dev with -ci.
+          sed -i 's/httpd-php:8\.3-\(dev\|ci\)/httpd-php:${{ matrix.php_version}}-ci/g' docker-compose.yml
+          # Report the difference.
+          git diff
+
+      - name: Start services with Docker Compose
+        run: docker compose up -d
+
+      - name: Install zip
+        run: |
+          docker compose exec -T web apt-get update -y
+          docker compose exec -T web apt-get install zip -y
+
+      - name: Update composer
+        run: docker compose exec -T web composer self-update --2
+
+      - name: Build dependencies
+        run: docker compose exec -T web composer require drupal/core:~${{ matrix.core_version }} drupal/core-composer-scaffold:~${{ matrix.core_version }} --update-with-all-dependencies --ansi --no-progress --no-interaction
+
+      - name: Run grumphp
+        run: docker compose exec -T web ./vendor/bin/grumphp run
+
+      - name: Components build
+        run: docker compose exec -T web bash ./scripts/build-ecl-pipeline.sh
+
+      - name: Create release archive
+        run: |
+          echo "Tag name from GITHUB_REF_NAME: $GITHUB_REF_NAME"
+          echo "Tag name from github.ref_name: ${{  github.ref_name }}"
```

Comment:
Did you notice any difference between the two expressions?
Or do you suspect they could be different in some scenarios?
Otherwise, I'd say the env var $GITHUB_REF_NAME is easier to write.